dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -129,31 +132,35 @@ public static void main(String[] arguments)
   }
 
   /**
-   * Finds jar file paths for Maven artifacts and their dependencies.
+   * Finds jar file paths for Maven artifacts and their transitive dependencies.
    *
    * @param artifacts Maven artifacts to check
    * @return list of absolute paths to jar files
    * @throws RepositoryException when there is a problem in retrieving jar files
    */
   public static ImmutableList<Path> artifactsToClasspath(List<Artifact> artifacts)
       throws RepositoryException {
-    
+
     LinkedListMultimap<Path, DependencyPath> multimap = artifactsToPaths(artifacts);
     return ImmutableList.copyOf(multimap.keySet());
   }
-  
-  
+
   // Multimap is a pain, maybe just use LinkedHashMap<Path, List<DependencyPath>>
   /**
-   * Finds jar file paths for Maven artifacts and their dependencies.
+   * Finds jar file paths and dependency paths for Maven artifacts and their transitive
+   * dependencies. When there are multiple versions of an artifact, the closest to the root
+   * (`artifacts` argument) is picked up in the same manner as Maven's dependency mediation.
```

Comment:
This isn't quite a full description of what this method does. In particular, it doesn't say that we include multiple paths to the same version of the artifact, which is sort of a weird thing to do so it's worth a comment.

And on that subject, why do we need multiple dependency paths for the artifact? Why isn't the first one enough? Especially since if the first one is removed from the graph, a different version of the artifact may be picked?